TotalCrossVM/src/tcvm/tcvm.c

```diff
@@ -764,6 +765,11 @@ TC_API TValue executeMethod(Context context, Method method, ...)
                   if (c == null)
                      goto throwClassNotFoundException;
                }
+               if (thisClass != c)
+               {
+                  thisClass = c;
```

Comment:
`thisClass` the last time I saw it was `OBJ_CLASS(regO[code->mtd.this_])`. I had not seen it being updated. And `c` is `OBJ_CLASS(regO[code->mtd.this_])`

I didn't see any changes in `regO` nor in `mtd.this_`.